Parser/pegen_errors.c

```diff
@@ -248,7 +248,11 @@ get_error_line_from_tokenizer_buffers(Parser *p, Py_ssize_t lineno)
     assert((p->tok->fp == NULL && p->tok->str != NULL) || p->tok->fp == stdin);
 
     char *cur_line = p->tok->fp_interactive ? p->tok->interactive_src_start : p->tok->str;
-    assert(cur_line != NULL);
+    if (cur_line == NULL) {
+        // We can reach this point if the tokenizer buffers for interactive source have not been
```

Comment:
Perhaps have an assert here that ensures we hit this only in interactive mode?

```suggestion
        assert(p->tok->fp_interactive);
        // We can reach this point if the tokenizer buffers for interactive source have not been
```